data_structures/circular_queue_using_binaric_file.cpp

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file
+ * @brief Implementation of a queue using a file.
+ * @details
+ * This program implements a queue using a file.
+ * It includes a template class `MyQueue` defining the queue interface,
+ * and a class `QueueFile` implementing the queue using a file as storage.
+ * * @algorithm
+ * The qoeoe include two pointers to the head and the tail of the qoueue
+ * And it also contain fstraem file there the data of the queue kept.
+ * When removing the value of the head indexOfFirst increase by one
+ * @author [Your Name](https://github.com/your_handle)
+ * @see <Related File(s)>
+ */
+
+#include <fstream>
+#include <iostream>
+#include <string>
+
+using namespace std;
+
+/**
+ * @interface MyQueue
+ * @brief A generic queue interface.
+ * @tparam T The type of elements stored in the queue.
+ */
+template <class T>
+class MyQueue {
+ public:
+    /// Clears the queue.
+    virtual void clear() = 0;
+
+    /// Removes and returns the front element of the queue.
+    virtual T dequeue() = 0;
+
+    /// Adds an element to the rear of the queue.
+    virtual void enqueue(T value) = 0;
+
+    /// Returns the front element of the queue without removing it.
+    virtual T front() = 0;
+
+    /// Checks if the queue is empty.
+    virtual bool isEmpty() = 0;
+
+    /// Destructor.
+    virtual ~MyQueue() {}
+};
+
+/**
+ * @class QueueFile
+ * @brief A queue implementation using a file as storage.
+ * @tparam T The type of elements stored in the queue.
+ */
+template <class T>
+class QueueFile : public MyQueue<T> {
+    int queueSize = 10;
+    int indexOfFirst = 0;
+    int indexOfLast = 0;
+    fstream data;
+
+    /// Checks if the queue is full.
+    bool isFull() { return indexOfFirst == indexOfLast + 1; }
+
+ public:
+    /**
+     * @brief Constructor for QueueFile.
+     * @param size The size of the queue.
+     */
+    QueueFile(int size = 100) : queueSize(size) {
+        data.open("queue.dat", ios::out);
+        T stam;
+        for (int i = 0; i < size; i++) {
+            data.write((char*)&stam, sizeof(T));
+        }
+        data.close();
+        data.open("queue.dat", ios::out | ios::in);
+        if (!data)
+            throw "error in opening";
+    }
+
+    /// Clears the queue.
+    void clear() override {
+        indexOfFirst = 0;
+        indexOfLast = 0;
+    }
+
+    /// Checks if the queue is empty.
+    bool isEmpty() override { return indexOfFirst == indexOfLast; }
+
+    /// Returns the front element of the queue without removing it.
+    T front() override {
+        if (this->isEmpty())
+            throw "underflow";
```

Comment:
@Panquesito7  do you think throwing errors here is alright?

Reply:
So @realstealthninja just for understanding, do you mean that in case of underflow, the front function return null?
Best Regards,
Chavi

Reply:
I'm asking that myself to panq.

Reply:
Thank you, I see,
But I see that @Panquesito7 is busy and doesn't answer.
So maybe I can change this to return null so we can merge the PR now?
What do you think @realstealthninja?